src/main.cpp

```diff
@@ -220,10 +212,24 @@ std::unique_ptr<CRollingBloomFilter> txn_recently_in_block GUARDED_BY(cs_recentR
 /** Number of preferable block download peers. */
 std::atomic<int> nPreferredDownload{0};
 
+} // anon namespace
+
+/** All pairs A->B, where A (or one of its ancestors) misses transactions, but B has transactions.
+ * Pruned nodes may have entries where B is missing data.
+ */
+std::multimap<CBlockIndex *, CBlockIndex *> mapBlocksUnlinked;
+
+/** Global flag to indicate we should check to see if there are
+ *  block/undo files that should be deleted.  Set on startup
+ *  or if we allocate more file space when we're in prune mode
+ */
+bool fCheckForPruning = false;
+
 /** Dirty block file entries. */
 std::set<int> setDirtyFileInfo GUARDED_BY(cs_main);
 
-} // anon namespace
+std::vector<CBlockFileInfo> vinfoBlockFile;
```

Comment:
Let's get all these globals into globals.cpp to ensure their proper construction order.  Esp. important for any std:: or guarded data structures to ensure that they aren't destructed after their guard, for example.